### PR TITLE
feat: Make more deviceinfo response fields accessible

### PR DESCRIPTION
**Pull request by rytilahti**
This will make most of the available payload data for P110 available through `PlugDeviceState` and `DeviceInfo`.

### CODE REVIEW

File: plugp100/responses/device_state.py [hunk 1]
```diff
@@ -81,28 +81,75 @@ def try_from_json(kwargs: dict[str, Any]) -> Either['LedStripDeviceState', Excep
...
+        self.oem_id = kwargs['oem_id']
         self.firmware_version = kwargs["fw_ver"]
         self.hardware_version = kwargs["hw_ver"]
         self.mac = kwargs["mac"]
         self.nickname = base64.b64decode(kwargs["nickname"]).decode("UTF-8")
         self.model = kwargs["model"]
         self.type = kwargs["type"]
         self.overheated = kwargs.get("overheated", False)
+        self.power_protection_status = kwargs["power_protection_status"]
+        self.ip = kwargs['ip']
```
Comment (rytilahti) [inline plugp100/responses/device_state.py:133]: There is a mixture of double and single quotes throughout the code base. @petretiandrea, have you considered using black to enforce the wanted code style?

File: plugp100/responses/device_state.py [hunk 2]
```diff
@@ -81,28 +81,75 @@ def try_from_json(kwargs: dict[str, Any]) -> Either['LedStripDeviceState', Excep
...
+    # location data
+    latitude: int
+    longitude: int
+    timezone: str
+    time_difference: int
+    language: str
+
+    # device state
+    power_protection_status: str
+    device_on: bool
```
Comment (petretiandrea) [inline plugp100/responses/device_state.py:110]: I consider `device_on` a property of a specific tapo product, for example the tapo HUB doesn't have device_on. So I think it is better to remove it from here. While other parameters you suggest are welcome, thanks

File: plugp100/responses/device_state.py (hunk 1 above)
Comment (petretiandrea) [reply plugp100/responses/device_state.py:133]: Yes, I will add it

File: plugp100/responses/device_state.py [hunk 3]
```diff
@@ -81,28 +81,75 @@ def try_from_json(kwargs: dict[str, Any]) -> Either['LedStripDeviceState', Excep
...
+        self.default_states = kwargs["default_states"]
+
+        self.latitude = kwargs["latitude"]
+        self.longitude = kwargs["longitude"]
+        self.timezone = kwargs["region"]
+        self.time_difference = kwargs["time_diff"]
+        self.language = kwargs["lang"]
+
+        # device state
+        self.device_on = kwargs["device_on"]
```
Comment (petretiandrea) [inline plugp100/responses/device_state.py:146]: Same of above

**Review by petretiandrea — CHANGES_REQUESTED** [review]

File: plugp100/responses/device_state.py (hunk 2 above)
Comment (rytilahti) [reply plugp100/responses/device_state.py:110]: Ok, I can change that. I suppose the `power_protection_status`, `on_time` and `auto_off*` fields are also specific to the plugs?  Just to be sure, these should go to the `PlugDeviceState` class too, right?

Comment (petretiandrea) [reply plugp100/responses/device_state.py:110]: I think yes

Comment (rytilahti) [reply plugp100/responses/device_state.py]: Done!

**Pull request merged**